src/IIIFPresentation/API/Features/Storage/Requests/UpsertCollection.cs

```diff
@@ -132,21 +152,19 @@ await dbContext.TrySaveCollection<PresentationCollection>(request.CustomerId, lo
         {
             return saveErrors;
         }
-
-        var total = await dbContext.Collections.CountAsync(
+            
+        var total = await dbContext.Hierarchy.CountAsync(
```

Comment:
Could we make the round trip to get total optional if we get a full `PageSize` back? If we ask for `PageSize` 20 and get 3 records we know the total is 3.

```cs
var items = dbContext
  .RetrieveCollectionItems(request.CustomerId, databaseCollection.Id)
  .Take(settings.PageSize);
if (items.Count < settings.PageSize)
{
  // there can't be more as we've asked for PageSize and got less 
  total = items.Count;
}
else
{
  // if we get PageSize back then there may be more in db
  total = await dbContext.Hierarchy.CountAsync(....
}
```